addons/point_of_sale/static/src/app/screens/receipt_screen/receipt/receipt.xml

```diff
@@ -217,7 +218,7 @@
                 x
                 <t t-esc="env.utils.formatCurrency(line.price_display_one)" />
                 <div class="price_display pos-receipt-right-align">
-                    <span t-esc="env.utils.formatCurrency(line.price_display, false)" />
+                    <span t-if="!(lineIsPartOfCombo(line.id) and line.price_display===0)" t-esc="env.utils.formatCurrency(line.price_display, false)" />
```

Comment:
Change the condition. I think what you want to convey is "if part of combo, it should not be the parent". Replace "line.price_display===0" with a proper "parent check".

Reply:
The idea is to not show combo children prices if they are 0

Reply:
My mistake. Let's keep that.